src/document/DocumentCommandHandlers.js

```diff
@@ -572,14 +572,23 @@ define(function (require, exports, module) {
      * @private
      * Ensures the suggested file name doesn't already exit.
      * @param {Directory} dir  The directory to use
-     * @param {string} baseFileName  The base to start with, "-n" will get appended to make unique
+     * @param {object} fileNameOptions  Properties about the filename that should be used. The object should contain these properties:
+     *   baseFileName: The base to start with, "-n" will get appended to make unique
+     *   extension: Optional extension to be used in the filename
      * @param {boolean} isFolder True if the suggestion is for a folder name
      * @return {$.Promise} a jQuery promise that will be resolved with a unique name starting with
      *   the given base name
      */
-    function _getUntitledFileSuggestion(dir, baseFileName, isFolder) {
-        var suggestedName   = baseFileName + "-" + _nextUntitledIndexToUse++,
-            deferred        = $.Deferred();
+    function _getUntitledFileSuggestion(dir, fileNameOptions, isFolder) {
+        var baseFileName = fileNameOptions.baseFileName,
+            extension    = fileNameOptions.extension ? fileNameOptions.extension.replace(/^\.?/, ".") : "";
+
+        if(isFolder) {
+            extension = "";
+        }
+
+        var suggestedName = baseFileName + "-" + _nextUntitledIndexToUse++ + extension,
```

Comment:
Just so it doesn't read like a mistake, let's wrap `_nextUntitledIndexToUse++` in `(...)`